x/ethbridge/handler.go

```diff
@@ -56,28 +39,30 @@ func NewHandler(
 
 // Handle a message to create a bridge claim
 func handleMsgCreateEthBridgeClaim(
-	ctx sdk.Context, cdc *codec.Codec, bridgeKeeper Keeper, msg MsgCreateEthBridgeClaim, sugaredLogger *zap.SugaredLogger,
+	ctx sdk.Context, cdc *codec.Codec, bridgeKeeper Keeper, msg MsgCreateEthBridgeClaim,
 ) (*sdk.Result, error) {
+
+	logger := bridgeKeeper.Logger(ctx)
 	var mutex = &sync.RWMutex{}
 	mutex.Lock()
 	defer mutex.Unlock()
 
-	status, err := bridgeKeeper.ProcessClaim(ctx, types.EthBridgeClaim(msg), sugaredLogger)
+	status, err := bridgeKeeper.ProcessClaim(ctx, types.EthBridgeClaim(msg))
 	if err != nil {
-		sugaredLogger.Errorw("bridge keeper failed to process claim.",
-			errorMessageKey, err.Error())
+		logger.Error("bridge keeper failed to process claim.",
+			"error:", err.Error())
 		return nil, err
 	}
 	if status.Text == oracle.SuccessStatusText {
-		if err = bridgeKeeper.ProcessSuccessfulClaim(ctx, status.FinalClaim, sugaredLogger); err != nil {
-			sugaredLogger.Errorw("bridge keeper failed to process successful claim.",
-				errorMessageKey, err.Error())
+		if err = bridgeKeeper.ProcessSuccessfulClaim(ctx, status.FinalClaim); err != nil {
```

Comment:
Does this logger format things in a JSON format? That was one of the motivations behind using the sugared logger.

Reply:
you can define any logger you want. It is defined here https://github.com/Sifchain/sifnode/blob/develop/app/app.go#L129. You can have plain text or json by default or define your own logger by implementing the interface.

Reply:
Here are some docs I recommend reading https://docs.tendermint.com/master/nodes/logging.html and here you can see the change you can make to define json.

Reply:
I read that doc and it did not mention JSON anywhere in it.

To use JSON, do I need to build a JSON object, stringify it, then log it out?

Reply:
The one that would be used on the current chain is https://github.com/go-kit/kit/blob/master/log/json_logger.go. You can change a  node that is currently running to json format. You may need to define string methods if they are not defined, but if zap was working so will this.  The new version of the sdk uses zerolog.